src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts

```diff
@@ -898,36 +901,35 @@ export class GeneComparisonToolComponent implements OnInit, AVI, OnDestroy {
     }
   }
 
-  getCircleSize(pval: number) {
+  getCircleSize(pval: number | undefined) {
+    // define min and max size of possible circles in pixels
+    const MIN_SIZE = 6;
+    const MAX_SIZE = 50;
+
+    // shouldn't be undefined but if it is, don't show a circle
+    if (pval === undefined)
+      return 0;
+
+    // if significance cutoff radio button selected and 
+    // p-Value > significance threshhold, don't show
     if (this.significanceThresholdActive && pval > this.significanceThreshold) {
       return 0;
     }
 
     const pValue = 1 - (this.nRoot(pval, 3) || 0);
-    const size = Math.round(100 * pValue * 0.44);
```

Comment:
100 * 0.44 = 44 (What was previously the max size)
Since the legend is 50px for the largest circle, let's just multiply the pValue by the MAX_SIZE for simplicity.